### PR TITLE
Android: Update mobile and TV to use game covers

### PR DESCRIPTION
**Pull request by zackhow**
Using covers should give a consistent look to dolphin's library. I also moved the settings row to the bottom for the TV interface. This is more in with what most TV apps do and looks cleaner.   It saves the covers in dolphin-emu/Cache/GameCovers/$gameId.png. Custom covers can be placed in the game directory with extension '.cover.png'.

First pass will try to use the disk's region/country to get the cover, if that fails, it attempts 'US', if that fails it attempts 'EN'. 

There were issues using libcurl from the C code so I implemented getting the covers in the Java code. First, https requests were failing with CURLE_SSL_CACERT_BADFILE. I didn't dig too far into it but I never found a fix for an NDK build of libcurl. Second, when getting the request, the actual process of getting the covers was quite slow, taking a 1-2 seconds per cover, not sure why on this. Either way, I think doing it in the Java is best, so the C code can leave GameFile::UseGameCovers to return false on Android OS.

cover
cover2
phonecovers
phonecovers2

@mahdihijazi @gwicks @degasus @JosJuice Thanks!

### CODE REVIEW

**Comment by JMC47** [comment]
@dolphin-emu-bot rebuild

**Comment by JMC47** [comment]
Works on my phone.  LGTM functionality wise.

**Comment by zackhow** [comment]
Fixed the region and added loading of a custom cover if it exists.

I forgot to mention doing this in Java was also due to SConfig not being loaded pre emulation. But yes, libcurl just has a lot of issues used with android on top of being really slow. My library of 8 games took 10 seconds to get the covers(defaulting lang to US, and using http instead of https). Whereas just using standard volley got them near instantly.

**Comment by JosJuice** [comment]
Alright, those fixes look good.

I think it would be best if we had a setting (off by default) for whether you want to download game covers, like in DolphinQt, so that people won't accidentally use up a lot of mobile data if they have many games. Also, as mentioned in the DolphinQt PR, GameTDB wants us to add a notice to the GUI that the game covers come from them. Putting that where we have the setting would make sense.

**Comment by mahdihijazi** [comment]
@zackhow I just aw your reply that you did use Volley, there is no need to add one more library for that. We already have Picasso which is way better with handling images in Android and for sure better than loading the images using libcurl because of the image caching, down sampling  and some other features that is build in the library. Will go through the PR in details later.

**Comment by zackhow** [comment]
@JosJuice From talks in the IRC, it seemed like it was okay to only use covers for Android for a few reasons. First Android users probably won't have a huge library like PC users. So with a cover being only 100kb-140kb it won't take up much space on the device or bandwidth to download. Even at 100 games(unrealistic for android outside of Shield with an external drive) it would only require around 13mb of space/bandwidth. I assume most users will probably only have around 10 games, thus a little over a megabyte. This download only happens once, since the covers are saved to Cache after downloading.  Also for the TV it matches the aesthetics of most other gaming apps and in the end they just look a lot better than what came before for both TV and mobile. I'll figure out a good way to put credit to GameTDB somewhere noticeable.

@mahdihijazi So I have swapped it over to using Picasso to get the covers. I've never used it before so let me know if there is a better way to do this.  Fun fact, don't ever try to use .into(new Target{}), it seems to garbage collect the object before the callback ever happens if it is not setup a certain way. Didn't realize there was an (imageView, callback) version until way later than I should.

**Comment by JMC47** [comment]
@dolphin-emu-bot rebuild

**Comment by BhaaLseN** [comment]
Mantano/Bookari asks whether it should go and download covers when I add new PDFs (or EPUBs) to the reader; perhaps we could/should do something similar. Either ask on the first occasion (which might be the first time a user adds a game, the next time a cover is needed and none is there, or simply because the setting is neither yes nor no) or on a per-game basis (as soon as the user adds a new one; but that might be annoying at some point).
Just some food for thought; not really a recommendation that it _absolutely must be_ done that way.

**Comment by zackhow** [comment]
Added credit to GameTDB in the main settings.

File: Source/Android/app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -135,6 +135,7 @@
     <string name="wiimote_speaker_description">Enable sound output through the speaker on a real Wiimote (DolphinBar required).</string>
     <string name="audio_stretch">Audio Stretching</string>
     <string name="audio_stretch_description">Stretches audio to reduce stuttering. Increases latency.</string>
+    <string name="gametdb_thanks">Thanks to GameTDB.com for providing Gamecube and Wii covers!</string>
```
Comment (slx7R4GDZM) [inline Source/Android/app/src/main/res/values/strings.xml]: GameCube

**Comment by jhaleit** [comment]
I really like this idea! I hope it gets through :)

**Comment by weihuoya** [comment]
`public static void loadGameBanner(ImageView imageView, GameFile gameFile)` 
this function will called many times when scroll `RecyclerView`, and will download cover again even already in downloading queue.

Game Id is region specific, it don't have to try other region.

if there isn't a cover for game, it don't have to try every time.

**Pull request merged**